python/instrumentation/openinference-instrumentation-haystack/tests/openinference/haystack/test_instrumentor.py

```diff
@@ -614,6 +615,62 @@ def test_openai_generator_llm_span_has_expected_attributes(
     assert isinstance(attributes.get(LLM_TOKEN_COUNT_TOTAL), int)
 
 
+@pytest.mark.parametrize(
+    "default_template, prompt_builder_inputs",
+    [
+        pytest.param(
+            "Where is {{ city }}?",
+            {"template_variables": {"city": "Munich"}},
+            id="default-template",
+        ),
+        pytest.param(
+            "What is the weather in {{ city }}?",
+            {
+                "template": "Where is {{ city }}?",  # overrides default template
+                "template_variables": {"city": "Munich"},
+            },
+            id="input-template-overrides-default-template",
+        ),
+        pytest.param(
+            "Where is {{ city }}?",
+            {
+                "city": "Munich",
+            },
+            id="input-kwarg-recorded-as-template-variable",
+        ),
+        pytest.param(
+            "Where is {{ city }}?",
+            {
+                "template_variables": {"city": "Munich"},  # overrides kwarg
+                "city": "Berlin",
+            },
+            id="input-template-variables-overrides-input-kwarg",
+        ),
+    ],
+)
+def test_prompt_builder_llm_span_has_expected_prompt_template_attributes(
+    default_template: Optional[str],
+    prompt_builder_inputs: Dict[str, Any],
+    in_memory_span_exporter: InMemorySpanExporter,
+    setup_haystack_instrumentation: Any,
+) -> None:
+    prompt_builder = PromptBuilder(template=default_template)
+    pipe = Pipeline()
+    pipe.add_component("prompt_builder", prompt_builder)
+    output = pipe.run({"prompt_builder": prompt_builder_inputs})
+    assert output == {"prompt_builder": {"prompt": "Where is Munich?"}}
+    spans = in_memory_span_exporter.get_finished_spans()
+    assert len(spans) == 2
+    span = spans[0]
+    attributes = dict(span.attributes or {})
+    assert attributes.get(OPENINFERENCE_SPAN_KIND) == "LLM"
+    assert attributes.get(LLM_PROMPT_TEMPLATE) == "Where is {{ city }}?"
+    assert isinstance(
+        prompt_template_variables_json := attributes.get(LLM_PROMPT_TEMPLATE_VARIABLES), str
+    )
+    assert json.loads(prompt_template_variables_json) == {"city": "Munich"}
+
```

Comment:
When possible, I would recommend that we account for all the attributes via simply `assert atttrbutes.pop(..., None) is not None` and ending up with `assert attributes == {}`